core/src/main/scala/workflows/Magnetar.scala

```diff
@@ -69,14 +68,14 @@ object Magnetar extends Workflow[Unit] {
 
   def destroy(d: Deployment, dc: Datacenter, ns: Namespace): WorkflowF[Unit] = {
     val sn = d.stackName
-    logToFile(d.id, s"removing policy from vault: ${vaultLoggingFields(sn, ns = ns.name, dcName = dc.name)}") >>
-    deletePolicyFromVault(d.stackName, ns.name) >>
-    logToFile(d.id, s"removing alerts from consul ${alerts.alertingKey(sn)}") >>
-    deleteAlertsFromConsul(d.stackName) >>
-    logToFile(d.id, s"removing discovery tables from consul ${routing.Discovery.consulDiscoveryKey(sn)}") >>
-    deleteDiscoveryInfoFromConsul(sn) >>
-    logToFile(d.id, s"instructing ${dc.name}'s scheduler to decommission ${sn}") >>
-    delete(dc,d) >>
+    logToFile(d.id, s"removing policy from vault: ${vaultLoggingFields(sn, ns = ns.name, dcName = dc.name)}") *>
+    deletePolicyFromVault(d.stackName, ns.name) *>
```

Comment:
Confirming this didn't rely on lazily building the `WorkflowF` as construction here is strict.

Reply:
It's all `Free` so should be fine